.github/workflows/issue-metrics.yml

```diff
@@ -0,0 +1,104 @@
+name: Weekly issue metrics
+on:
+  workflow_dispatch:
+
+permissions:
+  issues: write
+  pull-requests: read
+
+jobs:
+  build:
+    name: issue metrics
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get dates for last week
+      shell: bash
+      run: |
+
+        ##########################################################
+        Create report for the previous week
+        ##########################################################
+        # Get the current date
+        current_date=$(date +'%Y-%m-%d')
+
+        # Calculate the previous week
+        previous_date=$(date -d "$current_date -1 week" +'%Y-%m-%d')
+        
+        echo "$previous_date..$current_date"
+        echo "prev_week=$previous_date..$current_date" >> "$GITHUB_ENV"
+        
+        ##########################################################
+        Create report for the previous month
+        ##########################################################
+        
+        # Calculate the first day of the previous month
+        #first_day=$(date -d "last month" +%Y-%m-01)
+
+        # Calculate the last day of the previous month
+        #last_day=$(date -d "$first_day +1 month -1 day" +%Y-%m-%d)
+
+        #Set an environment variable with the date range
+        #echo "$first_day..$last_day"
+        #echo "last_month=$first_day..$last_day" >> "$GITHUB_ENV"
```

Comment:
Not used at the moment. I left it in as a reference if you wanted to update the workflow such that it'd fetch reports for the previous month instead of the previous week.